packages/cli/src/routeGeneration/routeGenerator.ts

```diff
@@ -179,7 +179,7 @@ export class RouteGenerator {
 
   private getRelativeImportPath(fileLocation: string) {
     fileLocation = fileLocation.replace(/.ts$/, ''); // no ts extension in import
-    return `./${path.relative(this.options.routesDir, fileLocation).replace(/\\/g, '/')}`;
+    return `./${path.relative(this.options.routesDir, fileLocation).replace(/\\/g, '/')}${this.options.esm ? '.js' : ''}`;
```

Comment:
We can't make this assumption. May be `.js`, but you can import  `.ts` directly.

Reply:
even if you import a `.ts` file you need to write `.js` 😕

Reply:
https://gist.github.com/sindresorhus/a39789f98801d908bbc7ff3ecc99d99c#can-i-import-esm-packages-in-my-typescript-project